core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3217,6 +3230,21 @@ object ZIO extends ZIOCompanionPlatformSpecific {
     mergeAllPar(all)(Option.empty[A])((acc, elem) => Some(acc.fold(elem)(f(_, elem)))).map(_.get)
   }
 
+  /**
+   * Reduces an `Iterable[IO]` to a single `IO`, working in up to `n` fibers in parallel.
```

Comment:
Perhaps worth mentioning that `f` must be commutative and associative?

Reply:
Yes 👍